subprojects/inline/src/test/java/org/mockitoinline/InOrderVerificationTest.java

```diff
@@ -0,0 +1,191 @@
+ */
+package org.mockitoinline;
+
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.times;
+
+import java.util.function.Consumer;
+
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.MockedStatic;
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.exceptions.misusing.NotAMockException;
+import org.mockito.exceptions.verification.VerificationInOrderFailure;
+import org.mockito.verification.VerificationMode;
+
+public class InOrderVerificationTest {
+
+    @Test
+    public void shouldVerifyStaticMethods() {
+        mockedStaticTest(mockedStatic -> {
+            // given
+            InOrder inOrder = inOrder(StaticContext.class);
+
+            // when
+            StaticContext.firstMethod();
+            StaticContext.secondMethod(0);
+
+            // then
+            inOrder.verify(mockedStatic, StaticContext::firstMethod);
+            inOrder.verify(mockedStatic, () -> StaticContext.secondMethod(0));
+        });
+    }
+
+    @Test
+    public void shouldVerifyStaticAndInstanceMethods() {
+        mockedStaticTest(mockedStatic -> {
+            // given
+            StaticContext mocked = mock(StaticContext.class);
+            InOrder inOrder = inOrder(mocked, StaticContext.class);
+
+            // when
+            StaticContext.firstMethod();
+            mocked.instanceMethod();
+            StaticContext.secondMethod(10);
+
+            // then
+            inOrder.verify(mockedStatic, StaticContext::firstMethod);
+            inOrder.verify(mocked).instanceMethod();
+            inOrder.verify(mockedStatic, () -> StaticContext.secondMethod(10));
+        });
+    }
+
+    @Test
+    public void shouldVerifyStaticMethodsWithSimpleAndWrapperModes() {
+        mockedStaticTest(mockedStatic -> {
+            // given
+            InOrder inOrder = inOrder(StaticContext.class);
+
+            // when
+            StaticContext.firstMethod();
+            StaticContext.firstMethod();
+            StaticContext.secondMethod(0);
+
+            // then
+            inOrder.verify(mockedStatic, StaticContext::firstMethod, times(2));
+            inOrder.verify(mockedStatic, () -> StaticContext.secondMethod(0), timeout(100).atLeastOnce());
+        });
+    }
+
+    @Test(expected = MockitoException.class)
+    public void shouldThrowExceptionWhenModeIsUnsupported() {
+        mockedStaticTest(mockedStatic -> {
+            // given
+            VerificationMode unsupportedMode = data -> { };
+            InOrder inOrder = inOrder(StaticContext.class);
+
+            // when
+            StaticContext.firstMethod();
+
+            // then
+            inOrder.verify(mockedStatic, StaticContext::firstMethod, unsupportedMode);
+        });
+    }
+
+    @Test(expected = VerificationInOrderFailure.class)
```

Comment:
Nit: do not use `@Test(expected)` and instead use `org.assertj.core.api.Assertions.assertThatThrownBy`. We recently migrated away from that in #2510 (I should probably check if we can enable ErrorProne to error on this).